contrib/inventory/ec2.py

```diff
@@ -530,6 +539,55 @@ def get_rds_instances_by_region(self, region):
                 error = "Looks like AWS RDS is down:\n%s" % e.message
             self.fail_with_error(error, 'getting RDS instances')
 
+    def get_db_clusters_by_region(self, region):
+        client = boto3.client('rds', region_name=region)
+        clusters = client.describe_db_clusters()["DBClusters"]
+        account_id = boto.connect_iam().get_user().arn.split(':')[4]
+        c_dict = {}
+        for c in clusters:
+            # remove these datetime objects as there is no serialisation to json
+            # currently in place and we don't need the data yet
+            if 'EarliestRestorableTime' in c:
+                del c['EarliestRestorableTime']
+            if 'LatestRestorableTime' in c:
+                del c['LatestRestorableTime']
+
+            matches_filter = False
+
+            try:
+                # arn:aws:rds:<region>:<account number>:<resourcetype>:<name>
+                tags = client.list_tags_for_resource(
+                    ResourceName='arn:aws:rds:' + region + ':' + account_id + ':cluster:' + c['DBClusterIdentifier'])
+                c['Tags'] = tags['TagList']
+
+                if self.ec2_instance_filters:
+                    for filter_key, filter_values in self.ec2_instance_filters.items():
+                        # get AWS tag key e.g. tag:env will be 'env'
+                        tag_name = filter_key.split(":", 1)[1]
+                        # Filter values is a list (if you put commas into the value it will be multiple items) - ignore the rest of the items for now
+                        matches_filter = any(d['Key'] == tag_name and d['Value'] == filter_values[0] for d in c['Tags'])
+
+                        if matches_filter:
+                            # print "Add DB to list"
+                            # it matches a filter, so stop looking for further matches
+                            break
+                            # else:
+                            # print "ignore this DB, filter out"
+
+            except Exception as e:
+                if e.message.find('DBInstanceNotFound') >= 0:
+                    # AWS RDS bug (2016-01-06) means deletion does not fully complete and leave an 'empty' cluster.
+                    # Ignore errors when trying to find tags for these
+                    pass
+
+            # ignore empty clusters caused by AWS bug
+            if len(c['DBClusterMembers']) == 0:
+                continue
+            elif matches_filter:
+                c_dict[c['DBClusterIdentifier']] = c
```

Comment:
Relating to my previous comment, instance filters are iterated through here https://github.com/ansible/ansible/pull/15609/files#diff-c2e371537339bbab5d9d0cc5d6cb7adcR564 and matches_filter is the result of testing each filter. Doesn't this mean only the last filter needs to match for the cluster to be included?